File: boj/b_16953.ts
function sol16953(a: number, b: number) {
  let result = Number.MAX_SAFE_INTEGER;
  const dfs = (n: number, c: number) => {
    if (n > b) return;
    if (n == b) return (result = Math.min(result, c));
    dfs(n * 2, c + 1), dfs(+("" + n + "1"), c + 1);
  };
  dfs(a, 1);
  console.log(result === Number.MAX_SAFE_INTEGER ? -1 : result);
}
sol16953(100, 40021);

File: boj/b_16953.py
import sys
sys.stdin = open("input.txt", "r") # 제거
input = sys.stdin.readline

a, b = map(int, input().split())
result = sys.maxsize
def dfs(n, c):
    global result
    if n > b:
        return
    if n == b:
        result = min(result, c)
        return
    dfs(n * 2, c + 1), dfs(int(str(n) + "1"), c + 1)
dfs(a, 1)
print(-1 if result == sys.maxsize else result)
